### PR TITLE
TEIID-5596 further refinements to the infinispan separation

### PR DESCRIPTION
**Pull request by shawkins**
Here's the follow on changes described in the other pull request.

### CODE REVIEW

File: cache-infinispan/pom.xml [hunk 1]
```diff
@@ -23,6 +22,16 @@
         <groupId>org.infinispan</groupId>
         <artifactId>infinispan-core</artifactId>
     </dependency>
+
```
Comment (rareddy) [inline cache-infinispan/pom.xml]: Why these dependencies?

File: cache-infinispan/pom.xml [hunk 2]
```diff
@@ -23,6 +22,16 @@
         <groupId>org.infinispan</groupId>
         <artifactId>infinispan-core</artifactId>
     </dependency>
+    
+    <dependency>
```
Comment (rareddy) [inline cache-infinispan/pom.xml]: Why these dependencies?

File: runtime/src/main/java/org/teiid/runtime/EmbeddedServer.java [hunk 3]
```diff
@@ -379,15 +379,35 @@ public Object invoke(Object proxy, Method method, Object[] args)
 		}
 		
 		if (config.getCacheFactory() == null) {
-            config.setCacheFactory(new CacheFactory() {
-                @Override
-                public <K, V> Cache<K, V> get(String name) {
-                    return new LocalCache<>(name, 256);
+		    String infinispanConfig = config.getInfinispanConfigFile();
```
Comment (rareddy) [inline runtime/src/main/java/org/teiid/runtime/EmbeddedServer.java]: I am trying to get rid of the inclusion of Infinispan directly or indirectly, put this in the hands of the developer of embedded to choose. I do not see much value in providing any reflective based impl except for testing. I feel this code needs to be in higher order frameworks like Thorntail or SB.

File: cache-infinispan/pom.xml (hunk 2 above)
Comment (shawkins) [reply cache-infinispan/pom.xml]: Notice that they were moved from runtime to cache.  They are transitive dependencies of infinispan.

File: runtime/src/main/java/org/teiid/runtime/EmbeddedServer.java (hunk 3 above)
Comment (shawkins) [reply runtime/src/main/java/org/teiid/runtime/EmbeddedServer.java]: > I am trying to get rid of the inclusion of Infinispan directly or indirectly, put this in the hands of the developer of embedded to choose.

You are still free to set a cachefactory any way you want.  

Infinispan is completely optional with this change.  It is also backwards compatible without a code change (just a pom change) and has an appropriate deprecation.  

My position is that it's not good to silently default to a non-concurrent cache implementation, so I'd prefer this change for now and we can refine it later.

**Pull request merged**